File: Models/UserAnswer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OnlineExam.Models
{
    [Table("UserAnswer")]

    public class UserAnswer
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [ForeignKey(nameof(Answer))]
        public int AnswerId { get; set; }
        public Answer Answer { get; set; }

        [ForeignKey(nameof(Question))]
        public int QuestionId { get; set; }
        public Question Question { get; set; }

        [ForeignKey(nameof(User))]
        public int UserId { get; set; }
        public User User { get; set; }
        public DateTime AnswerDate { get; set; }
    }
}

File: src/Models/UserAnswer.ts
import { Answer } from "./Answer";
import { Question } from "./Question";
import { User } from "./User";

export interface UserAnswer {
    id: number;
    answerId: number;
    answer?: Answer;
    questionId: number;
    question?: Question;
    userId: number;
    user?: User;
    answerDate: Date;
  }